### PR TITLE
add missing regex flags (x and m)

### LINKED ISSUE #279: Regex multiline mode (m flag)
Regex multiline mode (m flag) is required by SPARQL ( https://www.w3.org/TR/xpath-functions/#flags ) but not supported by RE2.

### LINKED ISSUE #280: Regex whitespace removal (x flag)
Regex whitespace removal (x flag) is required by SPARQL ( https://www.w3.org/TR/xpath-functions/#flags ) but not supported by RE2.

### PR DESCRIPTION
**Pull request by mcb5637**
close #279 
close #280

### CODE REVIEW

File: private/rdf4cpp/regex/RegexImpl.cpp [hunk 1]
```diff
@@ -20,9 +20,42 @@ static re2::RE2::Options translate_flags(Regex::flag_type const flags) {
...
+        x.append("(?m)");
+    }
+    if (flags.contains(RegexFlag::RemoveWhitespace)) {
+        uint64_t classes = 0;
+        char prev = '\0';
+        for (char const c : regex) {
+            if (c == '[' && prev != '\\') {
+                ++classes;
+            }
+            else if (c == ']' && prev != '\\') {
```
Comment (liss-h) [inline private/rdf4cpp/regex/RegexImpl.cpp]: minor formatting comment: this should be written as:

```c++
if (cond) {
    // stuff
} else if (cond2) {
    // stuff
}
```

File: private/rdf4cpp/regex/RegexImpl.cpp [hunk 2]
```diff
@@ -20,9 +20,42 @@ static re2::RE2::Options translate_flags(Regex::flag_type const flags) {
     return o;
 }
 
+static re2::RE2 build_regex(std::string_view regex, Regex::flag_type flags) {
```
Comment (liss-h) [inline private/rdf4cpp/regex/RegexImpl.cpp:23]: Please add a link to please add a direct link to the flags doc: https://www.w3.org/TR/xpath-functions/#flags

and state that RE2 does not support m and x by default

File: private/rdf4cpp/regex/RegexImpl.cpp [hunk 3]
```diff
@@ -20,9 +20,42 @@ static re2::RE2::Options translate_flags(Regex::flag_type const flags) {
...
+    auto opt = translate_flags(flags);
+    if (!flags.contains(RegexFlag::Multiline) && !flags.contains(RegexFlag::RemoveWhitespace)) {
+        return {regex, opt};
+    }
+    std::string x{};
+    x.reserve(regex.size()+4);
+    if (flags.contains(RegexFlag::Multiline)) {
+        x.append("(?m)");
```
Comment (liss-h) [inline private/rdf4cpp/regex/RegexImpl.cpp:34]: Is this one different from `Options::set_one_line`?

**Review by liss-h — CHANGES_REQUESTED** [review]
Thx for the PR, find my comments below

File: private/rdf4cpp/regex/RegexImpl.cpp (hunk 3 above)
Comment (mcb5637) [reply private/rdf4cpp/regex/RegexImpl.cpp:34]: `Options::set_one_line` does not work, if we use re2 in its current syntax mode.

Comment (liss-h) [reply private/rdf4cpp/regex/RegexImpl.cpp:34]: thats unfortunate, but is fine

**Review by liss-h — APPROVED** [review]

**Pull request merged**